rclcpp_components/cmake/rclcpp_components_add_node.cmake

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Create a target which builds a shared library and also creates a target
+# which builds an executable
+# :param target: the executable name
+# :type target: string
+# :param ARGN: the list of source files for executable and shared library
+# :type ARGN: list of strings
+#
+function(rclcpp_components_add_node)
+
+  cmake_parse_arguments(
+    PARSED_ARGS
+    "EXCLUDE"
+    "TARGET"
+    "SRCS"
```

Comment:
I think we can improve usage by removing the explicit use of `TARGETS` and `SRCS`. Also, for clarity I would use the same name for the `EXCLUDE_FROM_ALL` flag. E.g.

```cmake
cmake_parse_arguments(
  ARGS
  "EXCLUDE_FROM_ALL"
  ""
  ""
  ${ARGN}
)
list(GET ARG_UNPARSED_ARGUMENTS 0 ARG_TARGET)
list(REMOVE_AT ARG_UNPARSED_ARGUMENTS 0)
set(ARG_SRCS ${ARG_UNPARSED_ARGUMENTS)
```
Then the usage looks like `rclcpp_components_add_node(my_target EXCLUDE_FROM_ALL mysrc.cpp myothersrc.cpp)`

Perhaps there's a better way about it.